source/IAccessibleHandler.py

```diff
@@ -781,7 +784,8 @@ def processDestroyWinEvent(window,objectID,childID):
 	# so can't use generic focus correction. (#2695)
 	focus=api.getFocusObject()
 	from NVDAObjects.IAccessible.mscandui import BaseCandidateItem
-	if objectID==0 and childID==0 and isinstance(focus,BaseCandidateItem) and window==focus.windowHandle and not eventHandler.isPendingEvents("gainFocus"):
+	windowClassName=winUser.getClassName(window)
+	if childID==0 and isinstance(focus,BaseCandidateItem) and window==focus.windowHandle and not eventHandler.isPendingEvents("gainFocus"):

```

Comment:
I'm assuming there are objectIDs other than 0 you are seeing? can you mention what they are? Certainly negative objectIDs should never be supported here as they would be for non-client parts of the window such as the system menu or scroll bars. Were you seeing positive objectIDs?

Reply:
Object ID of candidate item is positive. Their window handle is also different from host application.